### PR TITLE
Soroush pdf

### PR DESCRIPTION
**Pull request by sahroush**

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -60,12 +61,16 @@ async def album(ctx, sub, *args):
...
+    else :
+        await pagify(bot, ctx, links, names)
 
 
 @bot.command(name='nhentai',
              help='posts the given sauce \n' +
-                  'nsfw is off in sfw channels unless +nsfw is used \n',
+                  'nsfw is off in sfw channels unless +nsfw is used \n' +
+                  'sends a pdf instead of an album when +pdf is used',
              usage="<source number> [+nsfw]")
```
Comment (ParsaAlizadeh) [inline bot.py]: missing [+pdf] in usage

File: libs/nhentai.py [hunk 2]
```diff
@@ -6,4 +6,4 @@ def fetch_hentai(sixdigit):
     links = []
     for i in range(douj.pages):
         links.append(douj[i])
-    return links, douj.name
+    return links, douj.name
```
Comment (ParsaAlizadeh) [inline libs/nhentai.py]: missing `\n` at the end of file

File: libs/reddit.py [hunk 3]
```diff
@@ -5,26 +5,26 @@ def makeUrl(afterID, sub):
...
+            ismed = ismedia(post['data']['url'] , include_gifs)
             if ismed and post['data']['over_18']:
                 nsfw += [[post['data']['title'], post['data']['url']]]
             elif ismed:
-                sfw += [[post['data']['title'], post['data']['url']]]
+                sfw += [[post['data']['title'],  post['data']['url']]]
     except:
         pass
-    return sfw, nsfw
+    return sfw, nsfw
```
Comment (ParsaAlizadeh) [inline libs/reddit.py]: missing `\n` at the end of file

File: libs/util.py [hunk 4]
```diff
@@ -111,3 +116,32 @@ async def react(reaction, user):
...
+                    images = []
+                    os.remove(name+".pdf")
+                    size_sum = 0
+    if(len(images) > 0):
+        images[0].save(name + '.pdf' ,save_all=True, append_images=images[1:])
+        await ctx.send(file=discord.File(name + '.pdf'))
+        for i in images:
+            i.close()
+        images = []
+        os.remove(name+".pdf")
```
Comment (ParsaAlizadeh) [inline libs/util.py:153]: and if `len(images) == 0`?

Comment (sahroush) [reply libs/util.py:153]: fixed

File: libs/reddit.py [hunk 5]
```diff
@@ -5,26 +5,33 @@ def makeUrl(afterID, sub):
...
+def ismedia(imageUrl , pdf):
+    ismed = ('.jpg' in imageUrl or
             '.webm' in imageUrl or
-            ('.gif' in imageUrl and include_gifs) or
+            ('.gif' in imageUrl and not pdf) or
             '.png' in imageUrl) and '.gifv' not in imageUrl
+    if(pdf):
+        response = requests.head(imageUrl, allow_redirects=True)
+        size = int(response.headers.get('content-length', -1))
+        if(size > size < 2250000):
```
Comment (ParsaAlizadeh) [inline libs/reddit.py]: What does L16 supposed to do?

**Review by ParsaAlizadeh — APPROVED** [review]

**Pull request merged**